File: src/main/python/IpAddrCount.py
import sys
from pyspark import SparkContext

if __name__ == "__main__":
  if len(sys.argv) != 3:
    print "Usage: IpAddrCount <file or directory> <ipAddr>"
    exit(0)
    
  logs = sys.argv[1]
  ipAddr = sys.argv[2]

  sc = SparkContext()
  count = sc.textFile(logs).filter(lambda line: ipAddr in line).count()
  print( "The Ip Address " + ipAddr + " appears " + str(count) + " times in the logs.")
  sc.stop()

File: src/main/scala/my/spark/examples/IpAddrCount.scala
package my.spark.examples

import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object IpAddrCount {
  def main(args: Array[String]) {

    if (args.length != 2) {
      System.err.println("Usage: IpAddrCount <directory|file> <ipAddr>")
      System.exit(1)
    }
    
    val logs = args(0)
    val ipAddr = args(1)

    val sc = new SparkContext()
    val count = sc.textFile(logs).filter(line => line.contains(ipAddr)).count()
    println( "The Ip Address " + ipAddr + " appears " + count + " times in the logs.")
    sc.stop()
  }
}
